### ISSUE TITLE
Profiles issues 

### ISSUE THREAD

**Issue by sana-ajani** [microsoft/vscode#139409]
After team discussion and brainstorming on this issue for user profiles https://github.com/microsoft/vscode/issues/116740, we will try to break this down into a few parts: 

1. Define UI, layouts, settings, extensions etc through profiles
2. Syncing profiles
3. Sharing profiles with other people

We can track all the explorations and issues that relate to the above in this one list:
-  ✅  Creating pre-defined layouts https://github.com/microsoft/vscode/issues/46851 - released in November. This can allow users to control and pre-determine settings they want to have as "defaults". Currently, Harvard CS50 uses a .devcontainer to outline these settings. 
    - [ ] Collect feedback from educators, verify if they would use this @sana-ajani 

- 💪 Exploration: Setting visibility for views/view containers and context menu items https://github.com/microsoft/vscode/issues/133787

- 💪 Exploration: Enabling extensions for specific projects https://github.com/microsoft/vscode/issues/40239

- Syncing multiple profiles https://github.com/microsoft/vscode/issues/92393

**Comment by Preposterone** [comment]
Thank you for picking up this functionality at last!

**Comment by sana-ajani** [comment]
Hi folks! The Profiles experience is an on-going feature we're refining, and we will post more of our team's discussions and brainstorming around this for each iteration in this issue moving forward. 

For June we are thinking about:
- **Profiles management across windows or workspaces.** 
    - Does it make sense to have Profiles per window rather than across multiple windows? 
    - What's the experience when managing different Profiles in different windows? 
    - Is there a use case for opening the same folder in different Profiles in different windows?
- **Preview or editing experience for Profiles**
    - What is the experience for when you import a new Profile? Do you want to edit the Profile or see a preview of it first?
    - How can you trust this Profile and its contents?

**Comment by alefragnani** [comment]
That's great to hear, specially seeing a few features I think to be a _must have_

>   * Does it make sense to have Profiles per window rather than across multiple windows?

Per window, as I will be able to work with different stacks (each one with its on settings/extensions) at the same time

>   * What's the experience when managing different Profiles in different windows?

The user could have a _default_ profile, and any new window will use it. If you open a folder with a Profile defined, apply it. In the end you may have (visually speaking) something like the [Peacock extension](https://marketplace.visualstudio.com/items?itemName=johnpapa.vscode-peacock) presents :-)

>   * Is there a use case for opening the same folder in different Profiles in different windows?

Not for me, but I suppose VS Code have a limitation to be solved first (https://github.com/microsoft/vscode/issues/2686)

>   * Should a workspace/folder remember the last Profile it was associated with?

For sure

>   * What is the experience for when you import a new Profile? Do you want to edit the Profile or see a preview of it first?

While importing, no at all, because "in my case", I'm not so interested in UI settings. But I see value on it for other users so, it would be interesting to have, for sure

About editing, totally. I even opened a feature request (https://github.com/microsoft/vscode/issues/148829) about it right after you first launched Profiles in Insiders, but it was closed as `extension-candidate` 😞. TBH, Forcing the development of a new extension simply because the team apparently forgot to use tabs and line breaks while saving the .code-profile file seems too much. 

>   * How can you trust this Profile and its contents?

I understand it could work similar to Workspace Trust, since settings and extensions are handled there too

Looking forward to the next releases

**Comment by cutsin** [comment]
> * Is there a use case for opening the same folder in different Profiles in different windows?

Yes, TypesScript project with monorepo, eg. react/deno's codes in one folder, they used different tsc

**Comment by sandy081** [comment]
Closing this as we are actively tracking issues separately.
